lib/open_telemetry/honeycomb/event.ex

```diff
@@ -0,0 +1,194 @@
+
+  ### Flattening
+
+  Map flattening uses periods (`.`) to delimit keys from nested maps, much like can be configured
+  on a dataset basis at the Honeycomb end. These span attributes before flattening:
+
+  ```elixir
+  %{
+    http: %{
+      host:  "localhost",
+      method: "POST",
+      path: "/api"
+    }
+  }
+  ```
+
+  ... becomes this event after flattening:
+
+  ```elixir
+  %{
+    "http.host" => "localhost",
+    "http.method" => "POST",
+    "http.path" => "/api",
+  }
+  ```
+  """
+
+  alias OpenTelemetry.Honeycomb.Cleaner
+  alias OpenTelemetry.Honeycomb.Config
+  alias OpenTelemetry.Honeycomb.Config.AttributeMap
+  alias OpenTelemetry.Honeycomb.Event.Extract
+  alias OpenTelemetry.Records.Span
+  require Record
+
+  @enforce_keys [:time, :data]
+  defstruct [:time, :data]
+
+  @typedoc """
+  Span attributes after flattening.
+
+  See [attribute limitations](#module-opentelemetry-honeycomb) for important detail on span
+  attribute names and values.
+
+  [attrlimits]: #module-opentelemetry-honeycomb
+  """
+  @type event_data :: %{optional(String.t()) => OpenTelemetry.attribute_value()}
+
+  @typedoc """
+  Honeycomb event suitable for POSTing to their batch API.
+
+  * `time`: ms since epoch; [MUST] be in ISO 8601 format, e.g. `"2019-05-17T09:55:12.622658Z"`
+  * `data`: `t:event_data/0` after flattening.
+
+  [MUST]: https://tools.ietf.org/html/rfc2119#section-1
+  """
+  @type t :: %__MODULE__{
+          time: String.t(),
+          data: event_data()
+        }
+
+  @doc """
+  The current UTC time in ISO 8601 format, e.g. `"2019-05-17T09:55:12.622658Z"`
+
+  Useful when creating events manually.
+  """
+  @spec now() :: String.t()
+  def now do
+    DateTime.utc_now() |> DateTime.to_iso8601()
+  end
+
+  @doc """
+  Convert one OpenTelemetry span to an event suitable for POSTing to the
+  [Honeycomb Events API][HCevents].
+
+  See [attribute limitations](#module-opentelemetry-honeycomb) for important detail on span a
+  attribute names and values.
+
+  [HCevents]: https://docs.honeycomb.io/api/events/
+  [HCstruct]: https://github.com/honeycombio/opentelemetry-exporter-go/blob/66047b9/honeycomb/honeycomb.go#L325
+  [HCdefs]: https://docs.honeycomb.io/working-with-your-data/managing-your-data/definitions/#tracing
+  """
+  @spec from_ot_span(
+          :opentelemetry.span(),
+          resource_attributes :: OpenTelemetry.attributes(),
+          attribute_map :: AttributeMap.t()
+        ) :: [t()]
+  def from_ot_span(ot_span, resource_attributes, attribute_map) do
+    span = Span.from(ot_span)
+
+    data =
+      span
+      |> Map.get(:attributes)
+      |> merge(resource_attributes)
+      |> merge(extracted_attributes(span, attribute_map))
+      |> Enum.into(%{})
+      |> Cleaner.clean()
+
+    time =
+      span.start_time
+      |> :opentelemetry.convert_timestamp(:microsecond)
+      |> DateTime.from_unix!(:microsecond)
+      |> DateTime.to_iso8601()
+
+    [%__MODULE__{time: time, data: data}]
```

Comment:
Why does this return a list with only one thing in it? Should it possibly return an empty list or a list with more than one struct in some circumstances?

Reply:
Honeycomb's data model doesn't have any concept matching OT events. So, we're going to have to emit extra Honeycomb records for those, just like Honeycomb's Go exporter. So, the return value of `Event.from_ot_span/3` will have to be be a list. So, I made it a list now.